cni-plugin/integration/flannel/flannel_test.go

```diff
@@ -0,0 +1,93 @@
+package flannel
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+)
+
+const (
+	ConfigDirectory = "/var/lib/rancher/k3s/agent/etc/cni/net.d"
+	FlannelConflist = "10-flannel.conflist"
+)
+
+func ls(dir string, t *testing.T) []string {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		t.Fatalf("failed to list files: %v", err)
+	}
+	fileNames := make([]string, len(files))
+	for i, f := range files {
+		fileNames[i] = f.Name()
+	}
+	return fileNames
+}
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+
+	return false
+}
+
+func TestMain(m *testing.M) {
+	runTests := flag.Bool("integration-tests", false, "must be provided to run the integration tests")
+	flag.Parse()
+
+	if !*runTests {
+		fmt.Fprintln(os.Stderr, "integration tests not enabled: enable with -integration-tests")
+		os.Exit(0)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestLinkerdCNIIsLastPlugin(t *testing.T) {
```

Comment:
Super nitpicky but CNIIS might be a bit hard to parse

```suggestion
func TestLinkerdIsLastCNIPlugin(t *testing.T)
```